classes/subscription/form/SubscriptionForm.inc.php

```diff
@@ -123,20 +126,47 @@ function readInputData() {
 		if (!$nonExpiring) {
 			// Start date is provided and is valid
 			$this->addCheck(new FormValidator($this, 'dateStart', 'required', 'manager.subscriptions.form.dateStartRequired'));
-			$this->addCheck(new FormValidatorCustom($this, 'dateStart', 'required', 'manager.subscriptions.form.dateStartValid', create_function('$dateStart', '$dateStartYear = strftime(\'%Y\', strtotime($dateStart)); $minYear = date(\'Y\') + SUBSCRIPTION_YEAR_OFFSET_PAST; $maxYear = date(\'Y\') + SUBSCRIPTION_YEAR_OFFSET_FUTURE; return ($dateStartYear >= $minYear && $dateStartYear <= $maxYear) ? true : false;')));
-			$this->addCheck(new FormValidatorCustom($this, 'dateStart', 'required', 'manager.subscriptions.form.dateStartValid', create_function('$dateStart', '$dateStartMonth = strftime(\'%m\', strtotime($dateStart)); return ($dateStartMonth >= 1 && $dateStartMonth <= 12) ? true : false;')));
-			$this->addCheck(new FormValidatorCustom($this, 'dateStart', 'required', 'manager.subscriptions.form.dateStartValid', create_function('$dateStart', '$dateStartDay = strftime(\'%d\', strtotime($dateStart)); return ($dateStartDay >= 1 && $dateStartDay <= 31) ? true : false;')));
+			$this->addCheck(new FormValidatorCustom($this, 'dateStart', 'required', 'manager.subscriptions.form.dateStartValid', function($dateStart) {
+				$dateStartYear = strftime('%Y', strtotime($dateStart));
+				$minYear = date('Y') + SUBSCRIPTION_YEAR_OFFSET_PAST;
+				$maxYear = date('Y') + SUBSCRIPTION_YEAR_OFFSET_FUTURE;
+				return ($dateStartYear >= $minYear && $dateStartYear <= $maxYear) ? true : false;
```

Comment:
I think here it could only be: return ($dateStartYear >= $minYear && $dateStartYear <= $maxYear) ? If so, then everywhere below too...